src/components/NewsletterSignup/NewsletterSignup.test.tsx

```diff
@@ -0,0 +1,332 @@
+import { sectionDataMap } from "../../helpers/types";
+
+// If you want to see what's happening, insert below render()
+// screen.debug();
+
+describe("NewsletterSignup Accessibility", () => {
+  const onSubmit = jest.fn(),
+    onChange = jest.fn(),
+    valueEmail = "";
+  it("Form state w/ all optional props (displayed and undisplayed) passes accessibility", async () => {
+    const { container } = render(
+      <NewsletterSignup
+        className={"my-class"}
+        id={"my-id"}
+        title="Testing"
+        formHelperText={"Form helper"}
+        onSubmit={onSubmit}
+        onChange={onChange}
+        valueEmail={valueEmail}
+      />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("Submitting state w/ all optional props (displayed and undisplayed) passes accessibility", async () => {
+    const { container } = render(
+      <NewsletterSignup
+        className={"my-class"}
+        id={"my-id"}
+        title="Testing"
+        formHelperText={"Form helper"}
+        onSubmit={onSubmit}
+        onChange={onChange}
+        valueEmail={valueEmail}
+        view={"submitting"}
+      />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("Error state w/ all optional props (displayed and undisplayed) passes accessibility", async () => {
+    const { container } = render(
+      <NewsletterSignup
+        className={"my-class"}
+        id={"my-id"}
+        title="Testing"
+        formHelperText={"Form helper."}
+        onSubmit={onSubmit}
+        onChange={onChange}
+        valueEmail={valueEmail}
+        view={"error"}
+      />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("Confirmation state w/ all optional props (displayed and undisplayed) passes accessibility", async () => {
+    const { container } = render(
+      <NewsletterSignup
+        className={"my-class"}
+        id={"my-id"}
+        title="Testing"
+        formHelperText={"Form helper"}
+        onSubmit={onSubmit}
+        onChange={onChange}
+        valueEmail={valueEmail}
+        view={"confirmation"}
+      />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("Bad email state w/ all optional props (displayed and undisplayed) passes accessibility", async () => {
+    const { container } = render(
+      <NewsletterSignup
+        className={"my-class"}
+        id={"my-id"}
+        title="Testing"
+        formHelperText={"Form helper"}
+        onSubmit={onSubmit}
+        onChange={onChange}
+        valueEmail={valueEmail}
+        view={"form"}
+        isInvalidEmail={true}
+      />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+});
+
+describe("NewsletterSignup Unit Tests", () => {
+  /** Notes
+   *
+   * 1. The newsletterSignupType tests are covered in the snapshot tests below.
+   * 2. Because the functionality of the submit click is handled entirely by the consuming app, there seems to be no
```

Comment:
This is true, but you can make tests for verifying that the functions you passed are being called. So you passed onSubmit and onChange and when you click on the submit button or type into the input field, you can verify that the jest mock functions are called.

Here is an example: https://github.com/NYPL/nypl-design-system/blob/74eb7820726b4d8720c5e9f43f25a3752b97b451/src/components/Checkbox/Checkbox.test.tsx#L193

Reply:
Added test for onChange and onSubmit